packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import { createClientRenderStrictMode } from 'test/utils';
+import { expect } from 'chai';
+import { DataGrid } from '@material-ui/data-grid';
+import { getColumnValues } from 'test/utils/helperFn';
+
+describe('<DataGrid /> - Filter', () => {
+  // TODO v5: replace with createClientRender
+  const render = createClientRenderStrictMode();
+
+  const baselineProps = {
+    rows: [
+      {
+        id: 0,
+        brand: 'Nike',
+      },
+      {
+        id: 1,
+        brand: 'Adidas',
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+      },
+    ],
+    columns: [{ field: 'brand' }],
+  };
+
+  let setProps: any;
+
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
+
+  const TestCase = (props: {
+    rows?: any[];
+    columns?: any[];
+    operator?: string;
+    value?: string;
+  }) => {
+    const { operator, value, rows, columns } = props;
+    return (
+      <div style={{ width: 300, height: 300 }}>
+        <DataGrid
+          columns={columns || baselineProps.columns}
+          rows={rows || baselineProps.rows}
+          filterModel={{
+            items: [
+              {
+                columnField: 'brand',
+                value,
+                operatorValue: operator,
+              },
+            ],
+          }}
+          disableColumnFilter={false}
+        />
+      </div>
+    );
+  };
+  function renderTestContainsA() {
+    const renderResult = render(<TestCase value={'a'} operator={'contains'} />);
+    setProps = renderResult.setProps;
+  }
+
+  it('should apply the filterModel prop correctly', () => {
+    renderTestContainsA();
```

Comment:
I think that the more duplicated the tests are, the better. I think we should run away from attractions as much as we can get away with tests, e.g. custom helper for a specific test. In this case, it's cheap, so I think it's a good win.

```suggestion
    render(<TestCase value={'a'} operator={'contains'} />);
```